frontend/src/pages/ArtifactDetails.tsx

```diff
@@ -14,30 +14,45 @@
  * limitations under the License.
  */
 
+import {
+  Api,
+  Artifact,
+  ArtifactCustomProperties,
+  ArtifactProperties,
+  GetArtifactsByIDRequest,
+  LineageView,
+  getResourceProperty,
+  titleCase,
+  LineageResource,
+} from 'frontend';
 import * as React from 'react';
 import { Page } from './Page';
 import { ToolbarProps } from '../components/Toolbar';
-import { RoutePage, RouteParams } from '../components/Router';
+import { RoutePage, RoutePageFactory, RouteParams } from '../components/Router';
 import { classes } from 'typestyle';
 import { commonCss, padding } from '../Css';
 import { CircularProgress } from '@material-ui/core';
-import { titleCase, getResourceProperty, serviceErrorToString } from '../lib/Utils';
 import { ResourceInfo, ResourceType } from '../components/ResourceInfo';
-import { Artifact } from '../generated/src/apis/metadata/metadata_store_pb';
-import { Apis, ArtifactProperties, ArtifactCustomProperties } from '../lib/Apis';
-import { GetArtifactsByIDRequest } from '../generated/src/apis/metadata/metadata_store_service_pb';
+import MD2Tabs from '../atoms/MD2Tabs';
+
+export enum ArtifactDetailsTab {
+  OVERVIEW = 0,
+  LINEAGE_EXPLORER = 1,
+}
+
+const tabs = {
+  [ArtifactDetailsTab.OVERVIEW]: { name: 'Overview' },
+  [ArtifactDetailsTab.LINEAGE_EXPLORER]: { name: 'Lineage Explorer' },
+};
+
+const tabNames = Object.values(tabs).map(tabConfig => tabConfig.name);
```

Comment:
I'm not sure on this one, it seems you are relying on `Object.values` iterating through `tabs` object by `0`, `1`, but that may not be guaranteed as far as I know: https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order/38218582#38218582.

I'd suggest making the order explicit.

Reply:
Updated, thanks for the reference!